engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -31,15 +36,17 @@
  *
  * @param <T> The type of the value this {@link SettingImpl} contains.
  */
-public class SettingImpl<T> implements Setting<T> {
+public final class SettingImpl<T> implements Setting<T> {
```

Comment:
I couldn't think of any reasons to extend this class in the future, but I may have been mistaken. The class was never designed to be extended in the first place. Having said that, however, I do agree with you. I think it may be better to leave the class non-final.

Reply:
Let's do that please.